src/Sarif.Viewer.VisualStudio.UnitTests/CodeAnalysisResultManagerTests.cs

```diff
@@ -0,0 +1,162 @@
+
+            IOpenFileDialog openFileDialog = this.mockOpenFileDialog.Object;
+
+            var mockCommonDialogFactory = new Mock<ICommonDialogFactory>();
+            mockCommonDialogFactory
+                .Setup(cdf => cdf.CreateOpenFileDialog())
+                .Returns(() => openFileDialog);
+
+            this.commonDialogFactory = mockCommonDialogFactory.Object;
+        }
+
+        [Fact]
+        public void CodeAnalysisResultManager_GetRebaselinedFileName_AcceptsMatchingFileNameFromUser()
+        {
+            // Arrange.
+            const string FileNameInLogFile = @"C:\Code\sarif-sdk\src\Sarif\Notes.cs";
+            const string RebaselinedFileName = @"D:\Users\John\source\sarif-sdk\src\Sarif\Notes.cs";
+
+            this.openFileDialogFileName = RebaselinedFileName;
+            this.openFileDialogResult = true;
+
+            var target = new CodeAnalysisResultManager(
+                this.fileSystem,
+                this.keyboard,
+                this.commonDialogFactory);
+
+            // Act.
+            string rebaselinedFileName = target.GetRebaselinedFileName(uriBaseId: null, fileName: FileNameInLogFile);
+
+            // Assert.
+            rebaselinedFileName.Should().Be(RebaselinedFileName);
+
+            Tuple<string, string>[] remappedPathPrefixes = target.GetRemappedPathPrefixes();
+            remappedPathPrefixes.Length.Should().Be(1);
+            remappedPathPrefixes[0].Item1.Should().Be(@"C:\Code");
+            remappedPathPrefixes[0].Item2.Should().Be(@"D:\Users\John\source");
+        }
+
+        [Fact]
+        public void CodeAnalysisResultManager_GetRebaselinedFileName_UsesExistingMapping()
+        {
+            // Arrange.
+            const string FirstFileNameInLogFile = @"C:\Code\sarif-sdk\src\Sarif\Notes.cs";
+            const string FirstRebaselinedFileName = @"D:\Users\John\source\sarif-sdk\src\Sarif\Notes.cs";
+
+            const string SecondFileNameInLogFile = @"C:\Code\sarif-sdk\src\Sarif.UnitTests\JsonTests.cs";
+            const string SecondRebaselinedFileName = @"D:\Users\John\source\sarif-sdk\src\Sarif.UnitTests\JsonTests.cs";
+
+            this.existingFiles.Add(SecondRebaselinedFileName);
+
+            this.openFileDialogFileName = FirstRebaselinedFileName;
+            this.openFileDialogResult = true;
+
+            var target = new CodeAnalysisResultManager(
+                this.fileSystem, 
+                this.keyboard,
+                this.commonDialogFactory);
+
+            // First, rebase a file to prime the list of mappings.
+            target.GetRebaselinedFileName(uriBaseId: null, fileName: FirstFileNameInLogFile);
+
+            // The first time, we prompt the user for the name of the file to rebaseline to.
+            this.mockOpenFileDialog.Verify(ofd => ofd.ShowDialog(), Times.Once());
+
+            // Now prepare to rebaseline a second file with the same prefix.
+            this.openFileDialogFileName = SecondRebaselinedFileName;
```

Comment:
@rtaket You're right that it's not required -- and then I don't need the variable `SecondRebaselinedFileName` at all. I'll make that change.

It's not actually any "safer", since without this assignment, `this.openFileDialogFileName` will still have its previous value, and so the mock dialog -- which uses that field -- would behave, if it were called, the same way as in the previous call. I think you're wondering whether it would be safer to set the value to `null`, in the hope of triggering an error if the dialog is erroneously shown. But since we do have an assertion which ensures that the dialog is shown only once, I won't bother with that. I'll just remove this line and the variable.

Reply:
@rtaket Oh, I still do need that variable, because I have to ensure the `File.Exists` returns `true` for that path; that's what prevents the dialog from being shown the second time. But I will remove the assignment to `openFileDialogFileName`.